content/en/docs/hertz/tutorials/toolkit/toolkit.md

```diff
@@ -0,0 +1,1019 @@
+2. Install hz:
+
+```bash
+go install github.com/cloudwego/hertz/cmd/hz@latest
+```
+
+3. Verify that the installation was successful `hz -v`, if the following version message is displayed, the installation was successful
+
+```console
+hz version v0.1.0
+```
+
+**Note**，Since hz creates soft links to its own binaries, make sure that the installation path of hz has writable permissions.
+
+## Operating mode
+
+To generate code using thrift or protobuf IDL, The corresponding compiler needs to be installed: [thriftgo](https://github.com/cloudwego/thriftgo) or [protoc](https://github.com/protocolbuffers/protobuf/releases).
+
+The code generated by hz, part of it is generated by the underlying compiler (usually about the structure defined in IDL), and the other part is the user-defined routing, method and other information in IDL. The user can run the code directly.
+
+In terms of execution flow, when hz uses thrift IDL to generate code, hz calls thriftgo to generate the go structure code and executes itself as a plug-in to thriftgo (named thrift-gen-hertz) to generate the rest of the code. This is also true when used with the protobuf IDL.
+
+```
+$> hz  ... --idl=IDL
+    |
+    | thrift-IDL
+    |---------> thriftgo --gen go:... -plugin=hertz:... IDL
+    |
+    | protobuf-IDL
+     ---------> protoc --hertz_out=... --hertz_opt=... IDL
+```
+
+How to install thriftgo/protoc:
+
+thriftgo:
+
+```
+$ GO111MODULE=on go install github.com/cloudwego/thriftgo@latest
+```
+
+protoc:
+
+```
+// brew installation
+$ brew install protobuf
+
+// Official image installation, using macos as an example
+$ wget https://github.com/protocolbuffers/protobuf/releases/download/v3.19.4/protoc-3.19.4-osx-x86_64.zip
+$ unzip protoc-3.19.4-osx-x86_64.zip
+$ cp bin/protoc /usr/local/bin/protoc
+// Make sure include/google goes under /usr/local/include
+$ cp -r include/google /usr/local/include/google
+```
+
+## Usage
+
+### Basic Usage
+
+#### new: Create a new Hertz project
+
+1.  Create a new project
+
+```bash
+// Execute under GOPATH, go mod name defaults to the current path relative to GOPATH, or you can specify your own
+hz new
+
+// Execute under non-GOPATH, you need to specify the go mod name
+hz new -mod hertz/demo
+
+// Tidy & get dependencies
+go mod tidy
+```
+
+When executed, it generates a scaffold for the Hertz project in the current directory.
+
+2. Compiling Projects
+
+```bash
+go build
+```
+
+3. Run the project and test it
+
+Run the project:
+
+```bash
+./{{your binary}}
+```
+
+Test:
+
+```bash
+curl 127.0.0.1:8888/ping
+```
+
+If it returns `{"message":"pong"}`, it means that the interface is connected.
+
+### Create a project based on thrift IDL
+
+#### new: Create a new project
+
+1. Create the thrift IDL file in the current directory
+
+```thrift
+// idl/hello.thrift
+namespace go hello.example
+
+struct HelloReq {
+    1: string Name (api.query="name"); // 添加 api 注解为方便进行参数绑定
```

Comment:
Add api annotations for easier parameter binding

Reply:
fixed.